web/packages/core/src/public/v1.ts

```diff
@@ -0,0 +1,173 @@
+import { MovieMetadata } from "../movie-metadata";
+import { DataLoadOptions, URLLoadOptions } from "../load-options";
+import { ReadyState } from "../internal/player/inner";
+
+export interface PlayerV1 {
+    /**
+     * A movie can communicate with the hosting page using fscommand
+     * as long as script access is allowed.
+     *
+     * @param command A string passed to the host application for any use.
+     * @param args A string passed to the host application for any use.
+     * @returns True if the command was handled.
+     */
+    onFSCommand: ((command: string, args: string) => boolean) | null;
+
+    /**
+     * Any configuration that should apply to this specific player.
+     * This will be defaulted with any global configuration.
+     */
+    config: URLLoadOptions | DataLoadOptions | object;
+
+    /**
+     * The effective config loaded with the last call to `load()`.
+     * If no such call has been made, this will be `null`.
+     */
+    readonly loadedConfig: URLLoadOptions | DataLoadOptions | null;
+
+    /**
+     * Indicates the readiness of the playing movie.
+     *
+     * @returns The `ReadyState` of the player.
+     */
+    get readyState(): ReadyState;
```

Comment:
We should probably use an actually public enum, maybe string based

Reply:
Probably, keep in mind the idea here may generally be to match https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/readyState#value (though HAVE_CURRENT_DATA, HAVE_FUTURE_DATA, and HAVE_ENOUGH_DATA don't necessarily all make sense for Ruffle, though depending how streamed loading of content is implemented maybe they will).

Reply:
Is this to match it for compatibility, or just for being a similar thing? Because if it's the latter; there's also document.readyState which _is_ a string, for precedence

Reply:
Context is in https://github.com/ruffle-rs/ruffle/pull/4084 and https://github.com/ruffle-rs/ruffle/pull/4813#issuecomment-892177759